src/effects.c

```diff
@@ -32,6 +32,16 @@ void set_effect_pos_offset(EffectInstance* effect, f32 x, f32 y, f32 z) {
     ((f32*)data)[3] = z;
 }
 
+void init_effect_data(void) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(gEffectTable); i++) {
+        s32 size = gEffectTable[i].dmaEnd - gEffectTable[i].dmaStart;
+        ASSERT_MSG(size <= 0x1000, "Effect 0x%x == 0x%x bytes (0x1000 limit)", i, size);
+    }
+    clear_effect_data();
+}
```

Comment:
IMO better off calling this `check_effect_sizes` or similar, and calling it `clear_effect_data` separately at the call site